### PR TITLE
[vfs] Better fix for NULL deref in File::Read()

### PR DESCRIPTION
**Pull request by Karlson2k**
Improve 775bab90089dd521408f8dc72434cfce63d51ebe

### CODE REVIEW

**Comment by fritsch** [comment]
@ace20022 what do you think?

jenkins build this please

**Comment by ace20022** [comment]
@fritsch the filesystem is not my speciality, sorry.

**Comment by kib** [comment]
@elepus ?

File: xbmc/filesystem/File.cpp [hunk 1]
```diff
@@ -536,6 +530,17 @@ ssize_t CFile::Read(void *lpBuf, size_t uiBufSize)
 
   try
   {
+    if (uiBufSize == 0)
+    {
+      // "test" read with zero size
+      // some VFSs don't handle correctly null buffer pointer
+      // provide valid buffer pointer for them
+      const int chunkSize = m_pFile->GetChunkSize();
+      auto_buffer dummyBuf(chunkSize > 1 ? (size_t)chunkSize : 1024);
```
Comment (arnova) [inline xbmc/filesystem/File.cpp]: Uh, why 1024 as a fallback? Seems like a random number (without a comment as to why).

**Comment by arnova** [comment]
@kib: You meant @elupus  ;-)

**Comment by MartijnKaijser** [comment]
ping if needed?

**Comment by Karlson2k** [comment]
Yep, still needed. Fix "test" read with internal buffer.

**Comment by MartijnKaijser** [comment]
If still needed please open a new PR and reference this one.

**Pull request closed without merging**